From 518c03355c29a204cbda8a041e5ab12b105d97f3 Mon Sep 17 00:00:00 2001
From: Koji Matsuoka <koji.matsuoka.xm@renesas.com>
Date: Thu, 7 Dec 2017 20:51:49 +0900
Subject: [PATCH 388/909] media: i2c: adv7604: Fix set_fmt function

commit 9250ef05dd971bb76a2c4a17354f1d91ea0e3331 from
git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-bsp.git

When calling the VIDIOC_G_CROP ioctl, the resolution value
is not reflected.
This patch adds resolution information query procedure in set_fmt()
for corresponding it.

Signed-off-by: Koji Matsuoka <koji.matsuoka.xm@renesas.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/media/i2c/adv7604.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/media/i2c/adv7604.c b/drivers/media/i2c/adv7604.c
index d13c1b10ceba..0e2ac20a9517 100644
--- a/drivers/media/i2c/adv7604.c
+++ b/drivers/media/i2c/adv7604.c
@@ -1966,6 +1966,11 @@ static int adv76xx_get_format(struct v4l2_subdev *sd,
 		fmt = v4l2_subdev_get_try_format(sd, cfg, format->pad);
 		format->format.code = fmt->code;
 	} else {
+		struct v4l2_dv_timings timings;
+
+		adv76xx_query_dv_timings(&state->sd, &timings);
+		state->timings = timings;
+
 		format->format.code = state->format->code;
 	}
 
-- 
2.17.1

